### PR TITLE
TECH-30764 Create testable lodgement

### PR DESCRIPTION
**Pull request by AliArmanLMG**
## The challenge being addressed:
It's very common that we get stuck in a testing cycle when trying to get over the hump with lenders getting confidence in our package. Also, creating loan applications to be able to generate Lixi packages for sending to lodgement targets could be time consuming and tedious. To make the process of sending lixi packages for testing purposes, lixi packages generated from loan applications based on different scenario like "First Home Owner" can be stored as Json files and each time that we want to send a test packages, we can desterilize those files and send the to the specified endpoint. Also this solution can be dockerized and shared with lenders to let them start calling the test endpoint by themselves and start receiving the test lixi packages by themselves. 

## Areas of impact:

- New endpoint called "PostTestPackage" has been introduced which send a lixi package based on the provided country and Scenario.
- The version of Dotnet has been updated from 6 to 8

## Risks, breaking change, actions not visible, external services / tools, etc.
If we created the endpoint in MyCRM_API to generate lixi packages based on real data, we could face security issues specially if the endpoint was supposed to be shared with lenders. By the proposed solution, there won't be any security risk because no real data is getting used also lenders should specify the endpoint for receiving the package by themselves which prevent them to send a test package to another lender by mistak.

### CODE REVIEW

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 1]
```diff
@@ -0,0 +1,1730 @@
...
+          }
+        ],
+        "BusinessChannel": {
+          "CompanyName": "TestBank",
+          "OtherIdentifier": "C9RZZZ",
+          "UniqueID": "Lender_5054",
+          "Contact": {
+            "WebAddress": "https://www.ubank.com.au/",
+            "OfficePhone": {
+              "Number": "02 9058 7402"
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:257]: is this a real tel number and web address, should we have a dummy tel number and web address here?

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 2]
```diff
@@ -0,0 +1,1730 @@
...
+            "ResidencyStatus": "Permanently in Australia",
+            "UnderstandApplication": true,
+            "UniqueID": "Family_8477374",
+            "x_Household": "Family_7102929",
+            "YearsInCurrentProfession": 6,
+            "Contact": {
+              "PreferredContact": "Mobile",
+              "EmailAddress": [
+                {
+                  "Email": "jeremy.kho+1111@loanmarketgroup.com",
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:974]: I think this should be a dummy email address here

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 3]
```diff
@@ -0,0 +1,1730 @@
...
+                  "EmailType": "Home"
+                }
+              ],
+              "Mobile": {
+                "AustralianDialingCode": "04",
+                "CountryCode": "61",
+                "Number": "0409503715"
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:981]: This should be a dummy mobile number here

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 4]
```diff
@@ -0,0 +1,1730 @@
...
+                    "GrossRegularOvertimeAmountConditionOfEmployment": false,
+                    "GrossSalaryAmount": 80000,
+                    "GrossSalaryFrequency": "Yearly",
+                    "WorkAllowanceAmountConditionOfEmployment": false
+                  }
+                }
+              }
+            ],
+            "PersonName": {
+              "FirstName": "Wanda",
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:1046]: Is this a real name? should it be changed to a dummy name.

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 5]
```diff
@@ -0,0 +1,1730 @@
...
+            "ProofOfIdentity": [
+              {
+                "CertifiedCopy": false,
+                "CountryOfIssue": "AU",
+                "DocumentNumber": "",
+                "DocumentType": "Other",
+                "ExpiryDate": "2027-09-23",
+                "IssuingOrganisation": "",
+                "MiddleNameOnDocument": "Wanda",
+                "NameOnDocument": "Wanda Maximoff",
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:1067]: Is this a real name? should it be changed to a dummy name.

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 6]
```diff
@@ -0,0 +1,1730 @@
...
+            "ResidencyStatus": "Permanently in Australia",
+            "UnderstandApplication": true,
+            "UniqueID": "Family_8477375",
+            "x_Household": "Family_7102929",
+            "YearsInCurrentProfession": 8,
+            "Contact": {
+              "PreferredContact": "Mobile",
+              "EmailAddress": [
+                {
+                  "Email": "jeremy.kho+2222@loanmarketgroup.com",
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:1098]: we should have a dummy email address here

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 7]
```diff
@@ -0,0 +1,1730 @@
...
+                  "EmailType": "Home"
+                }
+              ],
+              "Mobile": {
+                "AustralianDialingCode": "04",
+                "CountryCode": "61",
+                "Number": "0402510192"
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:1105]: we should have a dummy phone number here

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 8]
```diff
@@ -0,0 +1,1730 @@
...
+        "RelatedCompany": [
+          {
+            "BusinessName": "Australian Council of Educational Research",
+            "CompanyName": "Australian Council of Educational Research",
+            "UniqueID": "EmployerCompany_4688212",
+            "Contact": {
+              "x_Address": "Address_3510876",
+              "OfficePhone": {
+                "Number": "61404040404"
+              }
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:1496]: is this a dummy tel number?

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 9]
```diff
@@ -0,0 +1,1730 @@
...
+        ],
+        "SalesChannel": {
+          "Type": "Mortgage Broker",
+          "Aggregator": {
+            "ABN": "67632405504",
+            "LicenceNumber": "517192",
+            "LicenceType": "ACL",
+            "UniqueID": "Aggregator_2",
+            "Contact": {
+              "Email": "customerservice@loanmarket.com.au",
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:1564]: should these details be a dummy details?

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 10]
```diff
@@ -0,0 +1,1730 @@
...
+            "PersonName": {
+              "FirstName": "Testing",
+              "KnownAs": "Testing 1"
+            }
+          }
+        ],
+        "SalesChannel": {
+          "Type": "Mortgage Broker",
+          "Aggregator": {
+            "ABN": "67632405504",
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:1559]: should this ABN be a dummy ABN?

File: samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json [hunk 11]
```diff
@@ -0,0 +1,1730 @@
...
+            "CompanyName": "LoanMarket BDE Test",
+            "LicenceNumber": "517192"
+          },
+          "LoanWriter": {
+            "FirstName": "Jeremy",
+            "Surname": "Adviser Kho",
+            "UniqueID": "Family_6430652",
+            "Contact": {
+              "Email": "jeremy.kho@loanmarket.com.au",
+              "x_Address": "Address_3416383",
```
Comment (AndBeasley) [inline samples/MyCRM.Lodgement.Core/LixiPackageSamples/NewLoanApplication.json:1582]: all eth email and tel number values should be changed to dummy values.

**Review by AndBeasley — COMMENTED** [review]
left some questions around values in the sample lixi

**Comment by AliArmanLMG** [comment]
> left some questions around values in the sample lixi

it is obfuscated now. thanks

**Pull request merged**